dspace-api/src/main/java/org/dspace/discovery/SolrServiceImpl.java

```diff
@@ -453,6 +501,40 @@ public void updateIndex(Context context, boolean force) {
         }
     }
 
+    private void startMultiThreadIndex(Context context, boolean force, List<UUID> ids) throws SQLException {
+        int numThreads = ConfigurationManager.getIntProperty("discovery", "indexer.items.threads", 5);
+
+        if (ids == null) {
+            ids = new ArrayList<>();
+            Iterator<Item> items = itemService.findAllUnfiltered(context);
+            for (Item item : ImmutableList.copyOf(items)) {
+                ids.add(item.getID());
+            }
+
+            for (WorkspaceItem wsi : workspaceItemService.findAll(context)) {
+                ids.add(wsi.getItem().getID());
+            }
+
+            for (WorkflowItem wfi : workflowItemService.findAll(context)) {
+                ids.add(wfi.getItem().getID());
+            }
+        }
+        List<UUID>[] arrayIDList = Util.splitList(ids, numThreads);
```

Comment:
If this method is called upon multiple times from the process the number of threads that can be active at the same time are not limited to the "numThreads" variable. This could really lead to memory leaks where we have way too many threads running at the same time. It would really be best to use a spring TaskExecutor: https://docs.spring.io/spring/docs/4.3.x/spring-framework-reference/html/scheduling.html. This will ensure that the number of threads started by the discovery process is limited to the "numThreads" variable

Reply:
support for multi-threads has been withdrawn from this PR as suggested by @cwilper https://github.com/4Science/DSpace/pull/65#issuecomment-477177919 and will be eventually introduced in a separate PR/ticket